tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -4310,6 +4310,216 @@ TEST_F(OpConverterTest, ConvertConv3D) {
+          /*expected_output=*/{1,  2,  15, 3, 6,   -3, 22, 1,   88,
+                               56, 36, 1,  1, 105, 1,  16, -28, 1,
+                               42, 9,  3,  1, 7,   1,  11, 61,  5}},
+      // Basic - just 1x1 avg pooling - input = output, SAME padding
+      TestParams{
+          /*input_dims=*/{1, 3, 3, 3},
+          /*input=*/{1, 2,  15,  3, 6,  -3, 22, 1, 88, 56, 36, 1,  1, 105,
+                     1, 16, -28, 1, 42, 9,  3,  1, 7,  1,  11, 61, 5},
+          /*ksize=*/{1, 1, 1, 1, 1},
+          /*strides=*/{1, 1, 1, 1, 1},
+          /*padding=*/"VALID",
+          /*data_format=*/"NCDHW",
+          /*is_max_pooling=*/false,
+          /*expected_output_dims=*/{1, 3, 3, 3},
+          /*expected_output=*/{1,  2,  15, 3, 6,   -3, 22, 1,   88,
+                               56, 36, 1,  1, 105, 1,  16, -28, 1,
+                               42, 9,  3,  1, 7,   1,  11, 61,  5}},
+      // 3x3 max pooling
+      TestParams{
+          /*input_dims=*/{1, 3, 3, 3},
+          /*input=*/{1, 2,  15,  3, 6,  -3, 22, 1, 88, 56, 36, 1,  1, 105,
+                     1, 16, -28, 1, 42, 9,  3,  1, 7,  1,  11, 61, 5},
+          /*ksize=*/{1, 1, 3, 3, 3},
+          /*strides=*/{1, 1, 1, 1, 1},
+          /*padding=*/"VALID",
+          /*data_format=*/"NCDHW",
+          /*is_max_pooling=*/true,
+          /*expected_output_dims=*/{1, 1, 1, 1},
+          /*expected_output=*/{105}},
+      // 3x3 avg pooling
+      TestParams{
+          /*input_dims=*/{1, 3, 3, 3},
+          /*input=*/{1, 2,  15,  3, 6,  -3, 22, 1, 88, 56, 36, 1,  1, 105,
+                     1, 16, -28, 1, 42, 9,  3,  1, 7,  1,  11, 61, 5},
+          /*ksize=*/{1, 1, 3, 3, 3},
+          /*strides=*/{1, 1, 1, 1, 1},
+          /*padding=*/"VALID",
+          /*data_format=*/"NCDHW",
+          /*is_max_pooling=*/false,
+          /*expected_output_dims=*/{1, 1, 1, 1},
+          /*expected_output=*/{17.1851851851852}},
+      // 3x3 max pooling, NDHWC
+      TestParams{
+          /*input_dims=*/{3, 3, 3, 1},
+          /*input=*/{1, 2,  15,  3, 6,  -3, 22, 1, 88, 56, 36, 1,  1, 105,
+                     1, 16, -28, 1, 42, 9,  3,  1, 7,  1,  11, 61, 5},
+          /*ksize=*/{1, 3, 3, 3, 1},
+          /*strides=*/{1, 1, 1, 1, 1},
+          /*padding=*/"VALID",
+          /*data_format=*/"NDHWC",
+          /*is_max_pooling=*/true,
+          /*expected_output_dims=*/{1, 1, 1, 1},
+          /*expected_output=*/{105}},
+      // 3x3 avg pooling, NDHWC
+      TestParams{
+          /*input_dims=*/{3, 3, 3, 1},
+          /*input=*/{1, 2,  15,  3, 6,  -3, 22, 1, 88, 56, 36, 1,  1, 105,
+                     1, 16, -28, 1, 42, 9,  3,  1, 7,  1,  11, 61, 5},
+          /*ksize=*/{1, 3, 3, 3, 1},
+          /*strides=*/{1, 1, 1, 1, 1},
+          /*padding=*/"VALID",
+          /*data_format=*/"NDHWC",
+          /*is_max_pooling=*/false,
+          /*expected_output_dims=*/{1, 1, 1, 1},
+          /*expected_output=*/{17.1851851851852}},
```

Comment:
Can we make the input divisible by 27? Otherwise this can make the test fail easily due to numerical issues. Alternatively we can allow errors below instead of `ElementsAreArray` below.

Reply:
Addressed in https://github.com/tensorflow/tensorflow/pull/32843/commits/ab48f4961cc0e3fb31085b6a888dd38cbc9e80fa .